pkg/quota/admission/runonceduration/api/types.go

```diff
@@ -0,0 +1,25 @@
+package api
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// RunOnceDurationConfig is the configuration for the RunOnceDuration plugin.
+// It specifies a default override value for ActiveDeadlineSeconds for a run-once pod.
+// The project that contains the pod may specify a different setting. That setting will
+// take precedence over the one configured for the plugin here.
+type RunOnceDurationConfig struct {
+	unversioned.TypeMeta
+
+	// ActiveDeadlineSecondsOverride is the value to set on containers of run-once pods
+	// Only a positive value is valid. Absence of a value means that the plugin
+	// won't make any changes to the pod
+	ActiveDeadlineSecondsOverride *int64
+
+	// AllowProjectAnnotationOverride indicates whether an override can be set in a project
+	// annotation. The project annotation override will take precedence over the globally configured
+	// override.
+	AllowProjectAnnotationOverride bool
```

Comment:
> would adding an Enabled bool ... work better?

Well, I think so. Actually I think it'd be a nice convention for all plugins... not unusual to have one main breaker switch for plugins. Might be a little late to introduce...

> It'd be nice to be able to say "enforce the global setting, prevent it from being changed on a per-project basis".

I hadn't thought about that scenario. Still, it strikes me as a "you ain't gonna need it" that complicates the logic a bit and the documentation / user understanding more than a bit.

I think I've said enough; up to you :)